### PR TITLE
[4.0] [RFC] Sidebar links to dropdown

### PR DESCRIPTION
**Pull request by todordev**
### Summary of Changes
This PR removes the sidebar from Content and moves the links to dropdown menu in the toolbar. You will be able to jump between views when select an item from that menu.

It is not full implementation. It's only request for comment. If you like the idea, we can implement it on every view.

Before
before

After
after

### CODE REVIEW

File: layouts/joomla/toolbar/submenu.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+$options    = array_key_exists('options', $displayData) ? $displayData['options'] : [];
+$label      = array_key_exists('label', $options) ? $options['label'] : 'Submenu';
+
+$option     = Factory::getApplication()->input->get('option');
+$elementId  = 'js-submenu-'.$option;
+
+HTMLHelper::_('jquery.framework');
+
+Factory::getDocument()->addScriptDeclaration('
+	jQuery(document).ready(function($) {
```
Comment (C-Lodder) [inline layouts/joomla/toolbar/submenu.php:29]: Please rewrite this in vanilla JS and move to a separate file

File: layouts/joomla/toolbar/submenu.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+$entries    = array_key_exists('entries', $displayData) ? $displayData['entries'] : [];
+$options    = array_key_exists('options', $displayData) ? $displayData['options'] : [];
+$label      = array_key_exists('label', $options) ? $options['label'] : 'Submenu';
+
+$option     = Factory::getApplication()->input->get('option');
+$elementId  = 'js-submenu-'.$option;
+
+HTMLHelper::_('jquery.framework');
```
Comment (C-Lodder) [inline layouts/joomla/toolbar/submenu.php:26]: Remove this line once the below is altered

File: layouts/joomla/toolbar/submenu.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * @var array $displayData
+ */
+
+$entries    = array_key_exists('entries', $displayData) ? $displayData['entries'] : [];
+$options    = array_key_exists('options', $displayData) ? $displayData['options'] : [];
+$label      = array_key_exists('label', $options) ? $options['label'] : 'Submenu';
+
+$option     = Factory::getApplication()->input->get('option');
+$elementId  = 'js-submenu-'.$option;
```
Comment (C-Lodder) [inline layouts/joomla/toolbar/submenu.php:24]: space before and after the `.`

**Comment by brianteeman** [comment]
@C-Lodder better to agree on proceeding with the concept before commenting on the codestyle. No poiint wasting a contributors time if it doesnt get accepted. This is an rfc not a final proposal

**Comment by ReLater** [comment]
Why not simply remove the redundant (or am I wrong?) sidebar (red arrow)?

**Comment by ciar4n** [comment]
I like the idea. The dropdown might be good for a less aggressive change to removing the sidebar entirely.  A PR was actually merged to remove the sidebar from com_content however is somehow appears to have got reverted.. https://github.com/joomla/joomla-cms/pull/16386. A PR to remove the remaining sidebars has been open since June.. https://github.com/joomla/joomla-cms/pull/16909

**Comment by Fedik** [comment]
The idea is not bad, and frees a lot of space.

However it make thing more hidden and harder to find, 
and force me to do more than one click to reach destination.

I would prefer something like a "fake tabs" over a content table. Similar to what we had in Joomla 2.5. It still will free a lot of space and make easily reachable any View in ONE click.

**Comment by ciar4n** [comment]
It is a repeating conversation. Personally I don't agree with the argument of it been harder to find. As indicated in @ReLater screenshot this menu is usually an exact copy of the main menu 2nd level items. We have to get over the fact that every option can not be available with one click at all times, displaying the same options twice alongside each other is just silly.

**Comment by Fedik** [comment]
> We have to get over the fact that every option can not be available with one click at all times, displaying the same options twice alongside each other is just silly.

so you saying, moving mouse around is easily than just a click on immediately visible item?
and try to explain to average user which exactly drop down he/she should click to get destination, is a easy task?

I just saying that suggested changes are making complication in this direction.
Drop-down here can be a good idea for Mobile, but for Desktop better to be without.

I think Joomla! 1.5/2.5 had a good solution for this.

I wrote my opinion, and wrote why I think like that.
And I do not think that it's silly to make  life of average user more easy.

**Comment by C-Lodder** [comment]
@Fedik Think of it from 2 sides of the story. Yes, having the menu on the left make it easier for the user to navigate between views, however uses up a lot of space which you could say is another issue for users on smaller viewports.

Ciaran makes a good point when he says "_We have to get over the fact that every option can not be available with one click at all times_". He's spot on.

**Comment by Fedik** [comment]
I just leave it here:

screen 2018-02-04 19 05 21 814x208

it do not take much space

**Comment by mbabker** [comment]
No, it didn't take much space.  But, using the same sort of comparison, would you agree that the "CMS Sections" menu on https://developer.joomla.org/cms.html should also be duplicated as a second level menu under the "CMS" item on the main menu, and if it were should the menu items be accessible through both paths (the second level menu and a dedicated sidebar menu)?

Like was just said, we can't keep leaving 2 or 3 ways to do everything in the UI, even moreso the justification of retaining status quo isn't good enough.  It causes duplication, and depending on the implementation it wastes screen space.  A new template does not have to 100% mimic the old template, otherwise what's the point in building a new template if it's going to look and feel the exact same as the previous iteration just with a different color scheme?

This gets even more tricky because the customizable admin menus are now available.  These component sidebar or secondary menus do not read from that, they use a hardcoded list built in each component (typically in a helper class).  So now you're having inconsistent user experience if you're customizing the menus (potentially hiding something in that sidebar based on user group or view access level) and those reflections aren't consistently applied.

**Comment by brianteeman** [comment]
As @mbabker says whatever is done should support being customised.

**Comment by Fedik** [comment]
> But, using the same sort of comparison, would you agree that the "CMS Sections" menu on https://developer.joomla.org/cms.html should also be duplicated as a second level menu under the "CMS" item on the main menu, and if it were should the menu items be accessible through both paths (the second level menu and a dedicated sidebar menu)?

No, I do not agree, the "CMS Sections" is perfectly fine there, I see all of menu items in this section, and I can do a quick choose.

ok, I try to explain again, but it is last time 😄 

Let imagine, 
We made drop down (as suggested in the pull) or totally removed the duplication.

What is my actions to switch from **Article** view to **Category** view?:
1. Open Side menu or  drop-down (as suggested in the pull request)
2. Find Category item and click

Now imagine we made a tabs (as was in j2.5), my actions:
1. Find Category item and click

Easy and intuitive.

The duplication was made for quick access to a specific view. And if we move it to drop-down, then it loses all of sense. 
Then just drop it as @ReLater suggested.

But I would prefer to keep it as I have suggested. If it mimic something then let mimic. 
I do not really care how it will looks like. For me it more important how easy it is in a daily routine.

**Comment by Fedik** [comment]
> This gets even more tricky because the customizable admin menus are now available. These component sidebar or secondary menus do not read from that, they use a hardcoded list built in each component 

yeap, that is true

**Comment by C-Lodder** [comment]
@Fedik - Ciaran already submitted a prototype PR that uses tabs instead, however with the overall font-size in J4 being bigger than 1.5 and 2.5, the tabs will become stacked if there's not enough space, which will occurs with several components and/or multilang

**Comment by Fedik** [comment]
>  the tabs will become stacked if there's not enough space, which will occurs with several components and/or multilang

I have missed that
okay then

**Comment by ggppdk** [comment]
Surely saving space is needed

but navigation is at one side of screen
and then "sub-navigation" is ... at other side of screen ??

before sub-navigation was 1 action 
- move a little + click

after this PR 3 actions
- move to other side of (desktop) screen, (both my mouse and turn my eyes / head)
- click to open drop down
- click to select

i do not like it due to the above 
- plus it is "hidden" navigation, you need to know that there is something ther

if at least all navigation would stay at one side
for mobile it is good, but for desktop it is a pain

**Comment by dgrammatiko** [comment]
Can you please post an image of this page: `administrator/index.php?option=com_languages&view=overrides` with the open dropdown?

**Comment by samarthsharma36** [comment]
I have tested this item :white_check_mark: successfully on 1a48dd2686e026ab2bacf3e43f3a2a83376c6441<br /><br />Its a good idea to save space by adding drop-down menu but those options are very crucial ones, so it would be more user friendly if the UI for sidebar links stay as they are because this will be easier for the new users to get used to Joomla 4. A better idea would be a navigation bar which is suggested by @ITPrism in issue #19590 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19552">issues.joomla.org/tracker/joomla-cms/19552</a>.</sub>

**Comment by sanderpotjer** [comment]
@ITPrism thank you for the PR, but it is no longer needed. In the current 4.0-dev branch the sidebar is already removed (meanwhile).

@wilsonge @franz-wohlkoenig So this PR can be closed.

**Comment by wilsonge** [comment]
Thanks @sanderpotjer

**Pull request closed without merging**